hr_timesheet_sheet/models/hr_timesheet_sheet.py

```diff
@@ -0,0 +1,631 @@
+            'task_id': task and task.id or False,
+            'count_timesheets': len(timesheet),
+            'unit_amount': 0.0,
+        }
+        if self.id:
+            values.update({
+                'sheet_id': self.id,
+            })
+        if timesheet:
+            amount = sum([t.unit_amount for t in timesheet])
+            values.update({
+                'unit_amount': amount,
+            })
+        return values
+
+    @api.model
+    def _prepare_empty_analytic_line(self):
+        return {
+            'name': '/',
+            'employee_id': self.employee_id.id,
+            'date': self.date_start,
+            'project_id': self.add_line_project_id and
+            self.add_line_project_id.id or False,
+            'task_id': self.add_line_task_id and
+            self.add_line_task_id.id or False,
+            'sheet_id': self.id,
+            'unit_amount': 0.0,
+            'company_id': self.company_id.id,
+        }
+
+    @api.model
+    def add_line(self):
+        if self.add_line_project_id:
+            values = self._prepare_empty_analytic_line()
+            name_line = self._get_line_name(
+                self.add_line_project_id, self.add_line_task_id)
+            if self.line_ids.mapped('value_y'):
+                self.delete_empty_lines(False)
+            if name_line not in self.line_ids.mapped('value_y'):
+                self.timesheet_ids |= \
+                    self.env['account.analytic.line'].create(values)
+
+    def clean_timesheets(self, timesheet):
+        if self.id:
+            for aal in timesheet.filtered(lambda a: not a.sheet_id):
+                aal.write({'sheet_id': self.id})
+        repeated = timesheet.filtered(lambda t: t.name == "/")
+        if len(repeated) > 1:
+            timesheet = repeated.fusion_timesheets()
+        return timesheet
+
+    def delete_empty_lines(self, allow_empty_rows=False):
+        for name in self.line_ids.mapped('value_y'):
+            row = self.line_ids.filtered(lambda l: l.value_y == name)
+            if row:
+                task = row[0].task_id and row[0].task_id.id or False
+                ts_row = self.env['account.analytic.line'].search([
+                    ('project_id', '=', row[0].project_id.id),
+                    ('task_id', '=', task),
+                    ('date', '<=', self.date_end),
+                    ('date', '>=', self.date_start),
+                    ('employee_id', '=', self.employee_id.id),
+                    ('sheet_id', '=', self.id),
+                    ('company_id', '=', self.company_id.id),
+                ])
+                if allow_empty_rows and self.add_line_project_id:
+                    check = any([l.unit_amount for l in row])
+                else:
+                    check = not all([l.unit_amount for l in row])
+                if check:
+                    ts_row.filtered(
+                        lambda t: t.name == '/' and not t.unit_amount).unlink()
```

Comment:
Perhaps I think that it makes more sense to detect if any of the ts_row has name != '/' and then raise a Warning such as "You cannot delete analytic line XXXX for date YYYY because it has a specific description. Please remove the line from the "Details" section. 
@astirpe do you agree?

Reply:
@jbeficent not sure it's a good idea. I see the cleaning of empty lines as a silent mechanism. But maybe you are experiencing a use case different than mine.

What for sure can be done is to isolate the above two lines and put them in a separate method: this way we make it easier to extend and customize ..

Reply:
@astirpe the problem is when a user puts quantity 0 in the grid then saves, and the application still shows 3. He will not recognize that the application just ignored his command because the line happened to have a description different than '/'...

Reply:
Ok so I understand that you want to keep the mechanism as it is (row not deleted and unit_amount not updated) in order to guarantee some "safety" for the user who can accidentally put a unit amount to zero. In this case I agree with displaying a Warning.

Reply:
I just reproduced the example @jbeficent wrote. I agree that a warning in that case would be an usability improvement.
+1 for adding the warning.